### DISCUSSION TITLE
How to exactly use FindUnique with array (aka FindMany)

### DISCUSSION THREAD

**Discussion by BJladika** [prisma/prisma#7575] [Q&A] [answered]
Okay, so I have prisma model like that, aka student refers to a lot of points (1 to many?) like 1 User and a lot of his posts in prisma examples
```prisma
model umo_student {
  person_ptr_id Int             @id @default(autoincrement())
  student_id    String
  umo_person    umo_person      @relation(fields: [person_ptr_id], references: [id])
  umo_brspoints umo_brspoints[]

  @@index([student_id], name: "umo_student_student_id_48b6d86c")
}

model umo_brspoints {
  id             Int            @id @default(autoincrement())
  points         Float
  student_id     Int
  umo_student    umo_student    @relation(fields: [student_id], references: [person_ptr_id])

  @@index([student_id], name: "umo_brspoints_student_id_a6d8946a")
  @@index([points], name: "umo_brspoints_points_88fa519c")
}
```
graphql schema
```graphql
type Testing{ #Umo DB
  student_id: ID
  person_ptr_id: ID
  Points: [BRSPoint]
}

type BRSPoint{ #Umo DB
  id: ID!
  points: Float
  Discipline: Umo_Course
}
```
and resolvers
```javascript
  Testing:{ //Umo DB Resolver
    Points: (parent, _args, ctx) => {
      if (parent.person_ptr_id)
      return ctx.prisma2.umo_brspoints.findUnique({where: {student_id: parent.person_ptr_id}}).umo_brspoints()
      //return ctx.prisma2.umo_brspoints.findMany({where: {student_id: parent.person_ptr_id}})
      return null
    }
  },
```
And I'm getting error 
```
Invalid `prisma.umo_brspoints.findUnique()` invocation:\n\n{\n  where: {\n    student_id: 293\n    ~~~~~~~~~~\n  }\n}\n\nUnknown arg `student_id` in where.student_id for type umo_brspointsWhereUniqueInput. Available args:\n\ntype umo_brspointsWhereUniqueInput {\n  id?: Int\n}\n\n"
```
So what I'm doing wrong?

**Comment by ryands17** [comment, marked as answer]
As I'd mentioned before, `findUnique` cannot be used with a non-unique value. Here, `student_id` is not unique as it is a 1-many relation. You either need to use `findFirst` for just a single item or `findMany` for multiple items.

**Reply by BJladika** [reply]
But wait, how then Posts/Users works in example? I really don't get it, especially with a function like after findUnique **posts()**, is it not similar case? Maybe I need to change to id? smth like findUnique(id: parent.person_ptr_id).umo_brspoints()? Or what? Also what if I change `student_id` at `model umo_brspoints` to to unique or its techically can't be unique? I just can't use `findMany` because it'd cause N+1 query (which is not good)
```javascript
Author:{ 
posts: (parent, args, ctx) => {
      return ctx.prisma.user
        .findUnique({
          where: { id: parent.id },
        })
        .posts()
    },
  },
```

UPD: I tried to make student_id into `@unique` but I m now getting error that umo_brspoints() is not a fucntion `ctx.prisma2.umo_brspoints.findUnique(...).umo_brspoints is not a function",`

**Reply by ryands17** [reply]
As I said before, you can't query the same relation back with the fluent API. Your resolver needs to be written as:

```ts
Points: (parent, _args, ctx) => {
      if (parent.person_ptr_id)
        return ctx.prisma2.umo_student.findUnique({where: {person_ptr_id: parent.person_ptr_id}}).umo_brspoints()
     return null
}
```

This is how you can use `findUniqe`. On a unique field. You cannot call `umo_brspoints()` on the same relation `umo_brspoints`. The line below is not possible as both point to the same relation.

```ts
return ctx.prisma2.umo_brspoints.findUnique({where: {student_id: parent.person_ptr_id}}).umo_brspoints()
```

**Reply by BJladika** [reply]
So... whats a lesser evil? using `findMany` and getting n+1 query or using this fluent api with query such as
```
persons{
    id
    FIO
    Balls{
      Points{
        points
        Discipline{
          Details{
            Info{
              Name
            }
          }
        }
      }
    }
  }
}
```

**Reply by ryands17** [reply]
Use the fluent API wherever it is possible in the individual resolver.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
